tests/ui/rfcs/rfc-2008-non-exhaustive/improper_ctypes/auxiliary/types.rs

```diff
@@ -27,3 +27,14 @@ pub enum NonExhaustiveVariants {
     #[non_exhaustive] Tuple(u32),
     #[non_exhaustive] Struct { field: u32 }
 }
+
+// Note the absence of repr(C): it's not necessary, and recent C code can now use repr hints too.
```

Comment:
The docs don't make explicit that the ABI matches C's - https://doc.rust-lang.org/reference/type-layout.html#primitive-representation-of-field-less-enums - my sense is that's probably just an oversight?

But as-is it's not clear to me that this is accurate.

Reply:
```c
typedef enum Name: uint8_t {
    /* fields */
} Name;
```
is legal C code as-of C23 and must, by brute force of logic, match
```rust
#[repr(u8)]
enum Name {
    /* fields */
}
```
or fail to compile, according to my understanding.

Reply:
I *am* happy to revise the wording but
- it would be inconsistent with `improper_ctypes` as-is to warn on `u32` in FFI
- the inadequacy of the docs is a preexisting issue it seems
- if I had my druthers I would yeet the entire idea of linting on another crate's types in a crate that isn't the declaring crate from the lint, because it seems absurd, especially when in an `extern "C"` block that is only declaring a binding against C functions, not declaring a new Rust function (thus the bound function implicitly exists anyway), and this is very much the most-lamented lint.